packages/langium/src/lsp/type-hierarchy-provider.ts

```diff
@@ -99,7 +99,10 @@ export abstract class AbstractTypeHierarchyProvider implements TypeHierarchyProv
     }
 
     supertypes(params: TypeHierarchySupertypesParams, _cancelToken?: CancellationToken): TypeHierarchyItem[] | undefined {
-        const document = this.documents.getOrCreateDocument(URI.parse(params.item.uri));
+        const document = this.documents.getDocument(URI.parse(params.item.uri));
```

Comment:
LSP services can always be async. Shall we change the service API to support async calls?